reports/jshint-lines-of-shame.js

```diff
@@ -0,0 +1,48 @@
+// TODO: We need to figure out whether React will be supported natively for a reporter or if they
+// will be required to pre-compile
+
+// All subsequent files required by node with the extensions .es6, .es, .jsx and .js
+// will be transformed by babel.
+require('babel/register');
+
+var _ = require('lodash');
+var React = require('react');
+
+// TODO: This reporter requires multiple files, it may be better to put each reporter in a
+// directory. This also brings to light a potential issue with using the file name as the reporter
+// name. What if a reporter exports an Object like `{name: 'jsHint', process: function () {...}}`?
+var Report = require('./jshint-lines-of-shame/shame-report.jsx');
+
+var formatData = function formatData (rawData) {
+  return _.map(rawData, function(file) {
+    var errors = _.map(file.errors, function(error) {
+      return _.pick(error, ['reason', 'code', 'line', 'character']);
+    });
+    return {
+      path: file.file,
+      errors: errors
+    };
+  });
+};
+
+module.exports = function(results, cb) {
+  // TODO: This is a bit cumbersome
+  var jsHintResults = _.find(results, function(result) {
+    return result.config.name === 'jshint';
+  });
+
+  // TODO: This format necessitates specific knowledge/exploration of each test you're consuming
+  // Since each test will produce dramatically different output maybe an Object would be better than
+  // an Array. This would also solve the cumbersome name issue above.
+  var data = formatData(jsHintResults.logs.errors);
```

Comment:
My guess is that the majority of the reporters consuming this data will not be a simple loop over an array to consume the grade. They will most likely target one or more specific pieces of named data. I just thought an object could be nice since you wouldn't have to search for your reporter data and looping over an Object rather than an Array for the one reporter (badge) that does consume all reports the same way is no big deal. 
```
{
  jsHint: {
    config: {},
    grade: {},
    output: {
      
    }
  },
  jscs: {...}
  eslint: {...}
  jsInstpect: {...}
  rollbar: {...}
}
```